providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -120,18 +130,21 @@ int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
      * Otherwise it provides a padding oracle.
      */
     pad = buf[blocksize - 1];
+
+    /* pad is only checked for bad values so can still be tested in non-constant time */
     if (pad == 0 || pad > blocksize) {
         ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
         return 0;
     }
```

Comment:
I think this if should get removed. The for loop doesn't go to `pad` anymore, so there is no need to check it. The for loop will now give an error in case it's not valid. With this if, we'd leak it's not valid.

Reply:
Fixed.

Reply:
My comment was wrong. This check is needed, but it needs to be converted to constant time.